src/libraries/Microsoft.PowerFx.Connectors/Obsolete/ODataParameters.cs

```diff
@@ -72,90 +77,128 @@ public ODataParameters(ODataParameters oDataParameters)
+            AddTo(query);
+            return query;
+        }
+
+        public void AddTo(IDictionary<string, object> query)
+        {
             if (!string.IsNullOrEmpty(Compute))
             {
-                namedValues.Add(new NamedValue("$compute", FormulaValue.New(Compute)));
+                query["$compute"] = Compute;
             }
 
             if (Count)
             {
-                namedValues.Add(new NamedValue("$count", FormulaValue.New(true)));
+                query["$count"] = true;
             }
 
             if (!string.IsNullOrEmpty(Expand))
             {
-                namedValues.Add(new NamedValue("$expand", FormulaValue.New(Expand)));
+                query["$expand"] = Expand;
             }
 
             if (!string.IsNullOrEmpty(Filter))
             {
-                namedValues.Add(new NamedValue("$filter", FormulaValue.New(Filter)));
+                query["$filter"] = Filter;
             }
 
             if (!string.IsNullOrEmpty(Format))
             {
-                namedValues.Add(new NamedValue("$format", FormulaValue.New(Format)));
+                query["$format"] = Format;
             }
 
-            if (Index != 0)
+            if (Index != 0)            
             {
-                namedValues.Add(new NamedValue("$index", FormulaValue.New(Index)));
+                query["$index"] = Index;
             }
 
             if (Levels != 0)
             {
-                namedValues.Add(new NamedValue("$levels", FormulaValue.New(Levels)));
+                query["$levels"] = Levels;
             }
 
             if (!string.IsNullOrEmpty(OrderBy))
             {
-                namedValues.Add(new NamedValue("$orderby", FormulaValue.New(OrderBy)));
+                query["$orderby"] = OrderBy;
             }
 
             if (!string.IsNullOrEmpty(SchemaVersion))
             {
-                namedValues.Add(new NamedValue("$schemaversion", FormulaValue.New(SchemaVersion)));
+                query["$schemaversion"] = SchemaVersion;
             }
 
             if (!string.IsNullOrEmpty(Search))
             {
-                namedValues.Add(new NamedValue("$search", FormulaValue.New(Search)));
+                query["$search"] = Search;
             }
 
-            if (!string.IsNullOrEmpty(Select))
+            if (Select != null && Select.Count > 0)
             {
-                namedValues.Add(new NamedValue("$select", FormulaValue.New(Select)));
+                query["$select"] = Select;
             }
 
             if (Skip != 0)
             {
-                namedValues.Add(new NamedValue("$skip", FormulaValue.New(Skip)));
+                query["$skip"] = Skip;
             }
 
             if (Top != 0)
             {
-                namedValues.Add(new NamedValue("$top", FormulaValue.New(Top)));
+                query["$top"] = Top;
             }
-
-            return namedValues;
         }        
-
-        public Uri GetUri(Uri uriBase)
```

Comment:
#2411 - previously we'd get relative URLS here. 
NameValueCollection is not helpful here, so switching to dictionary.